Rubberduck.Parsing/VBA/DeclarationFinderManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Rubberduck.VBEditor.Application;
+using System;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+
+namespace Rubberduck.Parsing.VBA
+{
+    public class DeclarationFinderManager : IDeclarationFinderManager 
```

Comment:
I pondered myself whether to actually split this out. Let me explain the only reason why I did it in th end and also why I split out the `IDeclarationFinderProvider` interface. 
The  problem is that to refresh the `DeclarationFinder` a reference to the VBE, more precisely to the host application, is needed. So the sole purpose of this class is to remove the `ParseCoordinator`'s direct dependency on the host application. 
Wait, I just realized that the `RubberduckParserState` retains a reference to the VBE. That means that I can just add a method for refreshing the `DeclarationFinder` without specifying the host application. Then, this class can die again.